### PR TITLE
[DONE]: Translate `Your First Component`

### PR DESCRIPTION
**Pull request by eihabkhan**
- [x] Translate the "Your First Component" section
- [x] Translate the "Components: UI building blocks" section
- [x] Translate the "Defining a component" section
- [x] Translate the "Step 1: Export the component" section
- [x] Translate the "Step 2: Define the function" section
- [x] Translate the "Step 3: Add markup" section
- [x] Translate the "Using a component" section
- [x] Translate the "What the browser sees" section
- [x] Translate the "Nesting and organizing components" section
- [x] Translate the "Recap" section
- [x] Translate the "Challenges" section

### CODE REVIEW

**Comment by eihabkhan** [comment]
Done translating the page ✅

File: src/content/learn/your-first-component.md
Comment (A7med3bdulBaset) [inline src/content/learn/your-first-component.md:1]: Please review my commits

**Review by A7med3bdulBaset — APPROVED** [review]

**Comment by A7med3bdulBaset** [comment]
كالعادة ترجمة ممتازة
تم دمج الفرع
يمكنك الآن اختيار صفحة أخرى

File: src/content/learn/your-first-component.md
Comment (eihabkhan) [reply src/content/learn/your-first-component.md:1]: @A7med3bdulBaset Looks great, thanks for the corrections.

File: src/content/learn/your-first-component.md [hunk 1]
```diff
@@ -1,47 +1,47 @@
...
+يمثل هذا الترميز المرئي هذه المقالة `<article>`, عنوانها `<h1>`, وفهرس محتويات مختصر في شكل قائمة مرتبة `<ol>`. ترميز مرئي كهذا, مدمج مع CSS من أجل الأنماط التصميمية, و JavaScript من أجل التفاعلية, يكمن وراء كل شريط جانبي, صورة رمزية, نافذة, قائمة منسدلة - كل قطعة من واجهة مستخدم تراها في الويب.
 
-React lets you combine your markup, CSS, and JavaScript into custom "components", **reusable UI elements for your app.** The table of contents code you saw above could be turned into a `<TableOfContents />` component you could render on every page. Under the hood, it still uses the same HTML tags like `<article>`, `<h1>`, etc.
+تمكنك React من دمج الترميز المرئي, وCSS, وJavaScript في "مكونات" مخصصة, **عناصر واجهة المستخدم قابلة لإعادة الاستخدام لتطبيقك**. يمكن تحويل كود فهرس المحتوى الذي رأيته أعلاه إلى مكوّن `<TableOfContents />` الذي يمكن تصييره في كل صفحة. لا يزال يستخدم هذا المكوّن تحت الخطاء وسوم مثل `<article>`, `<h1>`, إلخ.
```
[the remark this answers is not in the document]
Comment (eihabkhan) [reply src/content/learn/your-first-component.md]: Noted 👌

**Pull request merged**